File: Programming Fundamentals with Python/Data Types and Variables/special_numbers.py

n = int(input())

for num in range(1, n+1):
    sum_digits = sum(int(digit) for digit in str(num))
    
    if sum_digits == 5 or sum_digits == 7 or sum_digits == 11:
        print(f"{num} -> True")
    else:
        print(f"{num} -> False")


"""
n = int(input())


for num in range(1, n+1, 1):
    digit = num
    sum_digits = 0
    
    while digit > 0:
        sum_digits += digit % 10
        digit = int(digit / 10)
    
    if sum_digits == 5 or sum_digits == 7 or sum_digits == 11:
        print(f"{num} -> True")
    else:
        print(f"{num} -> False")
"""

File: Programming Basics with Java/Nested Loops/SpecialNumbers.java
import java.util.Scanner;

public class SpecialNumbers
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
    
        int N = input.nextInt();
    
        for (int num = 1111; num <= 9999; num++) 
        {
            String numStr = Integer.toString(num);
            boolean isSpecial = true;
    
            for (char digitChar : numStr.toCharArray()) 
            {
                int digit = Character.getNumericValue(digitChar);
                if (digit == 0 || N % digit != 0) 
                {
                    isSpecial = false;
                    break;
                }
            }

            if (isSpecial)
            {
                System.out.print(num + " ");
            }
        }
    }
}
